src/Cli/dotnet/commands/dotnet-test/TestApplication.cs

```diff
@@ -61,15 +66,50 @@ public async Task RunHelpAsync()
                 FileName = isDll ?
                 Environment.ProcessPath :
                 _modulePath,
-                Arguments = BuildHelpArgs(isDll)
+                Arguments = BuildHelpArgs(isDll),
+                RedirectStandardOutput = true,
+                RedirectStandardError = true
             };
 
+            return await StartProcess(processStartInfo);
+        }
+
+        private async Task<int> StartProcess(ProcessStartInfo processStartInfo)
+        {
             if (VSTestTrace.TraceEnabled)
             {
                 VSTestTrace.SafeWriteTrace(() => $"Updated args: {processStartInfo.Arguments}");
             }
 
-            await Process.Start(processStartInfo).WaitForExitAsync();
+            var process = Process.Start(processStartInfo);
+            StoreOutputAndErrorData(process);
+            await process.WaitForExitAsync();
+
+            TestProcessExited.Invoke(this, new TestProcessExitEventArgs { OutputData = _outputData, ErrorData = _errorData, ExitCode = process.ExitCode });
```

Comment:
I think that the events could be null if nobody will subscribe or? `TestProcessExited?.Invoke...`